### PR TITLE
Multi-Node SLURM Support

### PR DESCRIPTION
**Pull request by dashstander**
Adds a `SlurmRunner` class for multi-node jobs SLURM clusters, without requiring OpenMPI or special permissions. Pairs with [this](https://github.com/EleutherAI/gpt-neox/pull/680) GPT-NeoX PR.

### CODE REVIEW

**Comment by StellaAthena** [comment]
@Quentin-Anthony or @ShivanshuPurohit can you review this and give it a thumbs up?

File: deepspeed/launcher/multinode_runner.py [hunk 1]
```diff
@@ -119,6 +120,65 @@ def get_cmd(self, environment, active_resources):
...
+    def get_cmd(self, environment, active_resources):
+        assert not getattr(self.args, 'detect_nvlink_pairs', False), "slurm backend does not support remapping visible devices"
+        total_process_count = sum(self.resource_pool.values())
+        srun_cmd = [
+            'srun',
+            '-n',
+            f'{total_process_count}',
+        ]
+
+        if getattr(self.args, 'slurm_comment', ''):
```
Comment (Quentin-Anthony) [inline deepspeed/launcher/multinode_runner.py:157]: Where do you define the `slurm_comment` arg?

Comment (Quentin-Anthony) [reply deepspeed/launcher/multinode_runner.py:157]: Ah nevermind. It's in https://github.com/EleutherAI/gpt-neox/pull/680

**Comment by Quentin-Anthony** [comment]
LGTM

**Review by Quentin-Anthony — APPROVED** [review]

**Comment by StellaAthena** [comment]
@Quentin-Anthony You're now a `CODEOWNER` of this repo. Can you resubmit your review and merge it?

**Review by Quentin-Anthony — APPROVED** [review]

**Pull request merged**